base/broadcast.jl

```diff
@@ -316,6 +311,20 @@ end
     end
 end
 @inline broadcast_c(f, ::Type{Any}, a...) = f(a...)
+broadcast_c(f, ::Type{Tuple}, As...) =
+    ntuple(k -> f(_tuplebroadcast_getargs(As, k)...), _tuplebroadcast_reslength(As))
+broadcast_c{T}(f, ::Type{Tuple}, ::Type{T}, As...) =
+    ntuple(k -> f(T, _tuplebroadcast_getargs(As, k)...), _tuplebroadcast_reslength(As))
+@inline _tuplebroadcast_getargs(::Tuple{}, k) = ()
+@inline _tuplebroadcast_getargs(As, k) =
+    (_broadcast_getindex(first(As), k), _tuplebroadcast_getargs(tail(As), k)...)
+@noinline _tuplebroadcast_reslength(As) =
```

Comment:
If inlining the length is the problem, then just defining

```julia
@noinline _tuplebroadcast_reslength(As) = length(broadcast_indices(As...)[1])
```

instead of these should do (it may be even faster).